### ISSUE TITLE
Investigate discrepancy in DSpace stats for September 2016

### ISSUE THREAD

**Issue by amandafrench** [VTUL/vtechworks#420]
I was compiling the stats for September by logging in and clicking Administrative --> Statistics as I always do, and I noticed that the September stats for item and bitstream views are lower by a factor of 10 than they usually are. 
![screenshot 2016-10-12 12 30 13](https://cloud.githubusercontent.com/assets/227453/19318686/ae5dbee0-9077-11e6-9d94-a088befa143b.png) 

When you get a chance, @keithgee, can you try to figure out why? See also the thread on Mattermost between me and @zoto724 and @alawvt.

**Comment by amandafrench** [comment]
I should add that although I've labeled this a bug, it could be that all our previous stats are wrong and September's are correct. However, @zoto724 says that server logs indicate about 800k bitstream downloads, so there's probably something weird with the DSpace stats-util, or else it's working differently because of the SSL and other changes.

**Comment by keithgee** [comment]
I've recovered September statistics. The report is available at https://vtechworks.lib.vt.edu/statistics?date=2016-9 when logged in as an administrator.

771,343 bitstream views
494, 531 item views

The stats that you were viewing were for 3 days, from September 28-September 30.

These are the legacy DSpace statistics, which aren't really supported anymore. They're built by scanning through the DSpace log files and matching on patterns that indicate bitstream downloads and so on.

I logged into the VTechWorks server, and the log files for most of September had been compressed with GZIP, probably by ITS to save space on the filesystem. I unzipped the files and rebuilt the statistics with the available logs using the DSpace stat-initial and the DSpace stat-report-initial command, available from the command-line "dspace" launcher.

There are some other issues here:
- there aren't any (or at least many) log files prior to September on the server. Because of this, monthly reports prior to September contain little useful information. Also, the cumulative report is very, very inaccurate as it contains only recent data. Reports are saved after they are generated, but when statistics are reconfigured and regenerated (as may have happened in recent sprints) the scripts expect all previous log files to be present so that reports can be created in the new format. @amandafrench may already have all of the previous statistics she needs in a spreadsheet. If not, we'll need to ask ITS to put log files back on the server so that legacy statistical reports can be rebuilt with previous information.

In a Mattermost conversation on January 11 it was mentioned that we'd like to keep VTechWorks log files on production for as long as possible, but this has been a while and it may be worth bringing this up again at the next DevOps meeting.

Also statistics related but less critical - at some point ITS added caching for bitstreams - perhaps on the F5 appliance. I don't know if DSpace is able to count these requests for cached bitstreams in its statistics. If it matters, we should also devote time to testing.

*\* Closing
